### PR TITLE
Security and Privacy section

### PR DESCRIPTION
**Pull request by igrigorik**
Preview: https://rawgit.com/w3c/resource-hints/security/index.html#security-and-privacy
- explain confidentiality risks
- explain interaction with CSP, Mixed Content, and Referrer
- explain site author best-practices

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -245,6 +245,28 @@
   </section>
 
   <section>
+    <h2>Security and Privacy</h2>
+
+    <p>The use of <a href="#dfn-resource-hint-link">resource hints</a> hints can expose new information about both the current page and user's activity on it. For example, observing the content of prefetched or prerendered resources, or observing the DNS, IP, and hostname information exposed during connection establishment, both for secure and plain-text connections, can reveal information about the page initiating such requests, and where dynamic resource hints are used, user's activity on it.</p>
```
Comment (yoavweiss) [inline index.html]: I'm not sure I understand what new information is exposed here (that cannot already be explicitly exposed via JS and XHRs).

File: index.html [hunk 2]
```diff
@@ -245,6 +245,28 @@
...
+
+    <p>To mitigate some of the above risks, the user agent MAY provide optional user setting to control the use of <a href="#dfn-resource-hint-link">resource hints</a>, and the user agent MUST:</p>
+
+    <ul>
+      <li>Treat <a>resource hint link</a>'s as <a href="https://w3c.github.io/webappsec/specs/mixedcontent/#category-optionally-blockable">Optionally-blockable Content</a>, as defined by [[MIXED-CONTENT]].</li>
+      <li>Enforce CSP policy restrictions for <a>resource hint link</a>'s, as defined by [[CSP2]].</li>
+      <li>Apply relevant the <a href="https://w3c.github.io/webappsec/specs/CSP2/#complications">Processing Complications recommendations</a> for optimistic resource fetching, as defined by [[CSP2]].</li>
```
Comment (yoavweiss) [inline index.html]: s/the//

**Comment by yoavweiss** [comment]
Looks good, other than the fact that I don't get what new info is exposed that can't be exposed by other means, so it'd be good to give concrete examples.

@mikewest - can you take a look?

File: index.html (hunk 1 above)
Comment (igrigorik) [inline index.html]: It's not that it's new, it's more about documenting the surface area, associated risks, and providing recommendations for how to account for these things... both on the UA implementation side, and for the site author.

Comment (yoavweiss) [inline index.html]: Then maybe s/can expose new information/can provide new means to expose information/

Comment (igrigorik) [inline index.html]: Makes sense, updated in 51be2cb - thanks!

**Comment by yoavweiss** [comment]
LGTM.

I guess we can land and ask the security folks for review later on.

**Comment by igrigorik** [comment]
@toddreifsteck @marcoscaceres can you guys please take a look at this one?

File: index.html [hunk 3]
```diff
@@ -245,6 +245,28 @@
   </section>
 
   <section>
+    <h2>Security and Privacy</h2>
+
+    <p>The use of <a href="#dfn-resource-hint-link">resource hints</a> can provide new means to expose information about both the current page and user's activity on it. For example, observing the content of prefetched or prerendered resources, or observing the DNS, IP, and hostname information exposed during connection establishment, both for secure and plain-text connections, can reveal information about the page initiating such requests, and where dynamic resource hints are used, user's activity on it.</p>
+
+    <p>To mitigate some of the above risks, the user agent MAY provide optional user setting to control the use of <a href="#dfn-resource-hint-link">resource hints</a>, and the user agent MUST:</p>
```
Comment (marcoscaceres) [inline index.html]: MAY and "optional" is tautological :) Drop the optional.

Comment (marcoscaceres) [inline index.html]: I think you want a full stop where the last comma is. It's a bit confusing you go from a MAY to a MUST.

File: index.html [hunk 4]
```diff
@@ -245,6 +245,28 @@
...
+
+    <ul>
+      <li>Treat <a>resource hint link</a>'s as <a href="https://w3c.github.io/webappsec/specs/mixedcontent/#category-optionally-blockable">Optionally-blockable Content</a>, as defined by [[MIXED-CONTENT]].</li>
+      <li>Enforce CSP policy restrictions for <a>resource hint link</a>'s, as defined by [[CSP2]].</li>
```
Comment (marcoscaceres) [inline index.html]: s/[[MIXED-CONTENT]]/[[!MIXED-CONTENT]]

Comment (marcoscaceres) [inline index.html]: s/[[CSP2]]/[[!CSP3]]

File: index.html [hunk 5]
```diff
@@ -245,6 +245,28 @@
...
+      <li>Apply relevant <a href="https://w3c.github.io/webappsec/specs/CSP2/#complications">Processing Complications recommendations</a> for optimistic resource fetching, as defined by [[CSP2]].</li>
+      <li>Apply relevant Referrer policy restrictions for <a>resource hint link</a>'s, as defined by [[REFERRER-POLICY]].</li>
```
Comment (marcoscaceres) [inline index.html]: As above, I think you should reference CSP3 instead.

File: index.html [hunk 6]
```diff
@@ -245,6 +245,28 @@
...
+
+    <p>To mitigate some of the above risks, the user agent MAY provide optional user setting to control the use of <a href="#dfn-resource-hint-link">resource hints</a>, and the user agent MUST:</p>
+
+    <ul>
+      <li>Treat <a>resource hint link</a>'s as <a href="https://w3c.github.io/webappsec/specs/mixedcontent/#category-optionally-blockable">Optionally-blockable Content</a>, as defined by [[MIXED-CONTENT]].</li>
+      <li>Enforce CSP policy restrictions for <a>resource hint link</a>'s, as defined by [[CSP2]].</li>
+      <li>Apply relevant <a href="https://w3c.github.io/webappsec/specs/CSP2/#complications">Processing Complications recommendations</a> for optimistic resource fetching, as defined by [[CSP2]].</li>
```
Comment (marcoscaceres) [inline index.html]: As above, I think you should reference CSP3 instead.

File: index.html [hunk 7]
```diff
@@ -245,6 +245,28 @@
...
+      <li>Apply relevant Referrer policy restrictions for <a>resource hint link</a>'s, as defined by [[REFERRER-POLICY]].</li>
+    </ul>
+
```
Comment (marcoscaceres) [inline index.html]: Make sure the links to specs in this section have a `[[!` to make them normative.

File: index.html [hunk 8]
```diff
@@ -245,6 +245,28 @@
...
+    <p>Additionally, the site author SHOULD take necessary precautions to specify the relevant [[MIXED-CONTENT]], [[CSP]], and [[REFERRER-POLICY]] policies to allow the user agent to register and apply them prior to processing a <a>resource hint link</a>:</p>
```
Comment (marcoscaceres) [inline index.html]: Conformance requirements for authors are not great. Maybe say "Additionally, site authors needs to take..."

File: index.html [hunk 9]
```diff
@@ -245,6 +245,28 @@
...
+
+    <ul>
+      <li>If <a>resource hint link</a> is provided via HTML markup, then the relevant `<meta>` policies should be declared ahead of them - e.g. <a href="http://www.w3.org/TR/CSP2/#delivery-html-meta-element">see `<meta>` recommendations</a> for CSP.</li>
```
Comment (marcoscaceres) [inline index.html]: should/SHOULD ?

File: index.html [hunk 10]
```diff
@@ -245,6 +245,28 @@
...
+      <li>If <a>resource hint link</a> is provided via the `Link` HTTP response header, then the relevant policies should also be delivered as an HTTP response header - e.g. see <a href="https://w3c.github.io/webappsec/specs/CSP2/#complications">Processing Complications</a> for CSP.</li>
```
Comment (marcoscaceres) [inline index.html]: Point to CSP3

File: index.html [hunk 11]
```diff
@@ -245,6 +245,28 @@
   </section>
 
   <section>
+    <h2>Security and Privacy</h2>
+
+    <p>The use of <a href="#dfn-resource-hint-link">resource hints</a> can provide new means to expose information about both the current page and user's activity on it. For example, observing the content of prefetched or prerendered resources, or observing the DNS, IP, and hostname information exposed during connection establishment, both for secure and plain-text connections, can reveal information about the page initiating such requests, and where dynamic resource hints are used, user's activity on it.</p>
+
```
Comment (marcoscaceres) [inline index.html]: This sounds kinda serious. Maybe a candidate for use only in TLS environments?

File: index.html (hunk 8 above)
Comment (marcoscaceres) [inline index.html]: I think you need to give more clear guidance to authors here as to how they do the above.

Comment (marcoscaceres) [inline index.html]: (actually, ignore the above).

File: index.html (hunk 11 above)
Comment (igrigorik) [inline index.html]: Note the "new means"... all of this is possible today by simply observing what resources the site is fetching and from which origins. I just want to be thorough in documenting the possible risks.

File: index.html (hunk 3 above)
Comment (igrigorik) [inline index.html]: Ack.

File: index.html (hunk 7 above)
Comment (igrigorik) [inline index.html]: Ack.

**Comment by igrigorik** [comment]
@marcoscaceres thanks for reviewing it! Updated, ptal.

**Pull request merged**